lib/dns.js

```diff
@@ -236,14 +250,17 @@ function resolver(bindingName) {
     req.hostname = name;
     req.oncomplete = onresolve;
     var err = binding(req, name);
-    if (err) throw errnoException(err, bindingName);
+    if (err) {
+      throw errnoException(err, bindingName);
+    }
+
     callback.immediately = true;
     return req;
   };
 }
 
 
-var resolveMap = {};
+var resolveMap = Object.create(null);
```

Comment:
Good question, we're using this object for a map, I thought about converting it to an ES2015 map but wasn't 100% sure about the implications or performance. `Object.create(null)` means that it can't be used as something other than a map - for example this guards against `__proto__` being passed as the second argument to `dns.resolve` which currently I assume is pretty harmless - but I'm not sure. 

With your blessing I'll refactor this to a `Map` gladly. Although that would make the above `export` lines double in size since it can be one-lined.